src/magneticField/CMZField.cpp

```diff
@@ -0,0 +1,458 @@
+// Poloidal Model (Model C) is taken from Katia Ferriere and Philippe Terral 2014 "Analytical models of X-shape magnetic fields in galactic halos"
+// Azimuthal Model is taken from M.Guenduez, J.B. Tjus, K.Ferriere, R.-J. Dettmar (2019)
+//                                                                                    
+CMZField::CMZField() {
+    useMCField = false;
+    useICField = true;
+    useNTFField = false;
+    useRadioArc = false;
+}
+
+double CMZField::scale(double x, double d) const{
+    return x*1./(pow(log(2),(1./d)));    
+}
+
+double CMZField::Br(double r, double z, double B1, double a, double L) const{//poloidal field
+    double r1=1/(1+a*pow(z,2.));
+    return 2*a*r*pow(r1,3.) *z *B1*exp(-r/L *r1);
+}
+
+double CMZField::Bz(double r, double z, double B1, double a, double L) const{//poloidal field
+    double r1=1/(1+a*pow(z,2.));
+    return pow(r1,2.) *B1*exp(-r/L *r1);
+}
+
+double CMZField::BrAz(double r, double phi, double z, double m, double B1, double eta, double R, double rr) const{//azimuthal field
+	double A;
+	if (r<rr)
+	{
+		A=  R/rr *(3.*r/rr -2.* pow(r/rr,2.));
+	}
+	else
+	{
+		A= R/r;
+	}
+	return B1 * exp(-pow(z/scale(R,2.),2.)) * cos(m/eta *log((r+1.)/(R+1.)) +m*phi) * A;
+}
+
+double CMZField::BphiAz(double r, double phi, double z, double m, double B1, double eta, double R, double rr) const{//azimuthal field
+	double A;
+	if (r<rr)
+	{
+		A=(1.+6.*(r-rr)/(2.*r-3.*rr) * ( tan(m/eta *log((r+1.)/(R+1.)) +m*phi) - sin(m/eta *log((r+1.)/(R+1.)))/cos(m/eta *log((r+1.)/(R+1.)) +m*phi) )) ;
+	}
+	else
+	{
+    	A= 1.;
+	}
+	return -1/eta *r/(r+1.) * BrAz(r,phi,z,m,B1,eta,R,rr) * A;
+}
+
+// azimuthal field in cartesian  coordinates// for molecular cloud region
+double CMZField::BxAz(double x, double y, double z, double m, double B1, double eta, double R, double rr) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    return (cos(phi)-1./eta * sin(phi)) *BrAz(r,phi,z,m,B1,eta,R,rr);
+}
+
+double CMZField::ByAz(double x, double y, double z, double m, double B1, double eta, double R, double rr) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    return (sin(phi)+1./eta*cos(phi))*BrAz(r,phi,z,m,B1,eta,R,rr);
+}
+
+double CMZField::BzAz(double x, double y, double z, double m, double B1, double eta, double R,double rr) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    return 0;//z component of the magnetic field is in molecular clouds zero
+}
+
+// Polodial field in cartesian coordinates// for non-thermal filament and the intercloud region
+double CMZField::ByPol(double x, double y, double z, double B1, double a1, double a2, double eta) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    double B2=B1/eta;// B1 is the observed magnetic field stremgth and eta the normalization factor otained from the integrations due to the normalization
```

Comment:
"B1 is the observed magnetic field stre**n**gth and eta the normalization factor otained from the integrations."